carmm/analyse/counterpoise_onepot.py

```diff
@@ -72,10 +76,9 @@ def counterpoise_calc(complex_struc, a_id, b_id, fhi_calc=None, a_name=None, b_n
             energies.append(energy_i)
         else:
             fhi_calc.template.outputname = species_list[index] + '.out'
-            properties = fhi_calc.implemented_properties
-            parameters = fhi_calc.parameters
-            parameters['ghosts'] = ghosts_lists_cp[index]
-            fhi_calc.template.update_parameters(properties, parameters)
+            fhi_calc.parameters['ghosts'] = ghosts_lists_cp[index]
+            # Scaled positions does not with empty sites.
+            fhi_calc.parameters['scaled'] = False
```

Comment:
In ase 3.23, `scaled` can be set in `parameters`. It is popped when preparing input files, so will not affect `control.in`

```
    def write_input(self, profile, directory, atoms, parameters, properties):
        """Write the geometry.in and control.in files for the calculation

        Parameters
        ----------
        directory : Path
            The working directory to store the input files.
        atoms : atoms.Atoms
            The atoms object to perform the calculation on.
        parameters: dict
            The parameters used to perform the calculation.
        properties: list of str
            The list of properties to calculate
        """
        parameters = self.update_parameters(properties, parameters)

        ghosts = parameters.pop('ghosts', None)
        geo_constrain = parameters.pop('geo_constrain', None)
        scaled = parameters.pop('scaled', None)
        write_velocities = parameters.pop('write_velocities', None)

        if scaled is None:
            scaled = np.all(atoms.pbc) 
```

Reply:
Is this backwards compatible with v3.22 also?

Reply:
Yes, the modification above only affects the v3.23 clause.
For ase v3.22, the scaled positions are disabled inside the `calculate_energy_ghost_compatible ` function.